src/content/developers/tutorials/erc721-vyper-annotated-code/index.md

```diff
@@ -0,0 +1,731 @@
+authorized by the owner, can do that.
+
+```python
+### MINT & BURN FUNCTIONS ###
+
+@external
+def mint(_to: address, _tokenId: uint256) -> bool:
+```
+
+This function always returns `True`, because if the the operation fails it is reverted.
+
+```python
+    """
+    @dev Function to mint tokens
+         Throws if `msg.sender` is not the minter.
+         Throws if `_to` is zero address.
+         Throws if `_tokenId` is owned by someone.
+    @param _to The address that will receive the minted tokens.
+    @param _tokenId The token id to mint.
+    @return A boolean that indicates if the operation was successful.
+    """
+    # Throws if `msg.sender` is not the minter
+    assert msg.sender == self.minter
+```
+
+Only the minter (the account that created the ERC-721 contract) can mint new tokens. This can be a 
+problem in the future if we want to change the minter's identity. In
+a production contract you would probably want a function that allows the minter to transfer
+minter priviliges to somebody else.
+
+```python
+    # Throws if `_to` is zero address
+    assert _to != ZERO_ADDRESS
+    # Add NFT. Throws if `_tokenId` is owned by someone
+    self._addTokenTo(_to, _tokenId)
+    log Transfer(ZERO_ADDRESS, _to, _tokenId)
+    return True
+```
+
+By convention, the minting of new tokens counts as a transfer from address zero.
+
+```python
+
+@external
+def burn(_tokenId: uint256):
+    """
+    @dev Burns a specific ERC721 token.
+         Throws unless `msg.sender` is the current owner, an authorized operator, or the approved
+         address for this NFT.
+         Throws if `_tokenId` is not a valid NFT.
+    @param _tokenId uint256 id of the ERC721 token to be burned.
+    """
+    # Check requirements
+    assert self._isApprovedOrOwner(msg.sender, _tokenId)
+    owner: address = self.idToOwner[_tokenId]
+    # Throws if `_tokenId` is not a valid NFT
+    assert owner != ZERO_ADDRESS
+    self._clearApproval(owner, _tokenId)
+    self._removeTokenFrom(owner, _tokenId)
+    log Transfer(owner, ZERO_ADDRESS, _tokenId)
+```    
+
+Anybody who is allowed to transfer a token is allowed to burn it. While a burn appears equivalent to
+transfer to the zero address, the zero address does not actually receives the token. This allows us to
+free up all the storage that was used for the token, which can reduce the gas cost of the transaction.
+
+
+# Using this Contract {#using-contract}
+
+In contrast to Solidty, Vyper does not have inheritence. This is a deliberate design choice to make the 
+code clearer and therefore easier to secure. So to create your own Vyper ERC-721 contract you take [this
+contract](https://github.com/vyperlang/vyper/blob/master/examples/tokens/ERC721.vy) and modify it 
+to implement the business logic you want.
+
+
+# Conclusion {#conclusion}
+
+
+For review, here are some of the most important ideas in this contract (in my opinion, yours is likely to vary):
```

Comment:
`(in my opinion, yours is likely to vary)`
hah, don't be afraid to own it with confidence =) it's your post